src/Symfony/Component/Messenger/Middleware/SendMessageMiddleware.php

```diff
@@ -47,21 +48,24 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
             // it's a received message, do not send it back
             $this->logger?->info('Received message {class}', $context);
         } else {
-            $shouldDispatchEvent = true;
             $senders = $this->sendersLocator->getSenders($envelope);
             $senders = \is_array($senders) ? $senders : iterator_to_array($senders);
-            foreach ($senders as $alias => $sender) {
-                if (null !== $this->eventDispatcher && $shouldDispatchEvent) {
-                    $event = new SendMessageToTransportsEvent($envelope, $senders);
-                    $this->eventDispatcher->dispatch($event);
-                    $envelope = $event->getEnvelope();
-                    $shouldDispatchEvent = false;
-                }
```

Comment:
This is a minor improvement:

1. The event was [originally dispatched outside the loop](https://github.com/symfony/symfony/pull/61843).
2. It was later [moved inside the loop](https://github.com/symfony/symfony/pull/30676) because of the retry mechanism.
3. [The retry mechanism was later extracted](https://github.com/symfony/symfony/pull/34185), so there's no longer a need for the event to be dispatched inside the loop.